### PR TITLE
filter identical h2h markets

### PR DESCRIPTION
**Pull request by andrewdamelio**
https://github.com/AugurProject/turbo/issues/636

### CODE REVIEW

**Comment by bthaile** [comment]
filter out in contract-calls.ts so the same market isn't loaded each block.

**Review by bthaile — CHANGES_REQUESTED** [review]
should do this logic in market loading and filter out of data.markets

**Comment by andrewdamelio** [comment]
> should do this logic in market loading and filter out of data.markets

👍 makes sense

**Comment by andrewdamelio** [comment]
@bthaile updated

File: packages/comps/src/utils/contract-calls.ts [hunk 1]
```diff
@@ -1552,7 +1552,20 @@ const retrieveMarkets = async (
...
   }
 
-  return { marketInfos, exchanges, blocknumber };
+  // If the target home spread is exactly zero then it will be set to 0.5, which is 5 in the contract.
+  // Filter out spread markets whose "value" (target home spread) is exactly 5,  Since this is identical to a head-to-head market.
+  const SPORTS_MARKET_TYPE_SPREAD = 1;
+  let marketInfosNoZeroSpread = marketInfos;
+  let exchangesNoZeroSpread = exchanges;
+
+  Object.keys(marketInfos).forEach((market) => {
```
Comment (bthaile) [inline packages/comps/src/utils/contract-calls.ts]: There is already a constant for spread.

File: packages/comps/src/utils/contract-calls.ts [hunk 2]
```diff
@@ -1552,7 +1552,20 @@ const retrieveMarkets = async (
...
+    if (marketInfos[market].sportsMarketType === SPORTS_MARKET_TYPE_SPREAD && marketInfos[market].spreadLine === 5) {
```
Comment (bthaile) [inline packages/comps/src/utils/contract-calls.ts]: Keep all the filtering and removing logic in on place “ getMarketInfos” is the place

**Review by bthaile — CHANGES_REQUESTED** [review]
Added comments

File: packages/comps/src/utils/contract-calls.ts [hunk 3]
```diff
@@ -1398,18 +1398,33 @@ export const getMarketInfos = async (
         .filter((id) => existingEvents.includes(marketInfos[id]?.eventId))
         .map((id) => Number(marketInfos[id]?.turboId));
 
-      addResolvedMarketToList(ignoreList, factoryAddress, [...ids, ...hiddenMarketsIds, ...existingEventIds]);
+      // If the target home spread is exactly zero then it will be set to 0.5, which is 5 in the contract.
+      const marketSpreadZeroIds = Object.keys(marketInfos)
+        .filter(
+          (id) =>
+            marketInfos[id].sportsMarketType === SPORTS_MARKET_TYPE.SPREAD &&
+            marketInfos[id].spreadLine === 5 &&
```
Comment (bthaile) [inline packages/comps/src/utils/contract-calls.ts]: "5" should prob be "0" when the line is zero

**Review by bthaile — CHANGES_REQUESTED** [review]
made a comment

**Review by bthaile — APPROVED** [review]
lgtm

**Pull request merged**